lib/rules/jsx-newline.js

```diff
@@ -22,41 +22,68 @@ module.exports = {
     fixable: 'code',
 
     messages: {
-      newLine: 'JSX element should start in a new line'
-    }
+      newLine: 'JSX element should start in a new line',
+      noNewLine: 'JSX elements should avoid new lines around them'
+    },
+
+    schema: [{
+      type: 'object',
+      properties: {
+        removeNewLines: {
+          default: false,
+          type: 'boolean'
+        }
+      }
+    }]
+
   },
   create(context) {
+    const configuration = context.options[0] || {};
     const jsxElementParents = new Set();
     const sourceCode = context.getSourceCode();
     return {
       'Program:exit'() {
         jsxElementParents.forEach((parent) => {
-          parent.children.forEach((element, index, elements) => {
-            if (element.type === 'JSXElement' || element.type === 'JSXExpressionContainer') {
-              const firstAdjacentSibling = elements[index + 1];
-              const secondAdjacentSibling = elements[index + 2];
-              if (
-                firstAdjacentSibling
+          if (configuration.removeNewLines) {
+            parent.children.forEach((element) => {
+              if (element.type === 'JSXText') {
```

Comment:
@ljharb I need some help here. My logic for detecting new lines was like iterating through the parents and for each child I am checking if they kind of match a new line. My concern is that the new lines will get counted multiple times as the parser moves through the nodes.
For something like this 
```
<div>

       <button></button>
       
</div>
```
I tried printing like for each parent what the new lines was and I got something like this
```
Parent identified
Newline found
Parent identified
Parent identified
Parent identified
Newline found
```
I am confused like how is the newline found after the last div instead it should be found after the last button right ?

Reply:
In your example, there's 5 newlines, but 2 blank lines - i think what you want to find is a *blank* line, ie, `\n\s*\n`, not a *newline*, which is `\n`?